modules/ROOT/pages/mule-error-concept.adoc

```diff
@@ -3,68 +3,163 @@ ifndef::env-site,env-github[]
+      cf-ray=50058b8add0a92fe-SJC
+   ]
+}
+  attributesMediaType=*/*
+}
 ----
++
+* `#[error.childErrors]` returns: `[]`
 
-== Error Types
+[[error_types]]
+== Mule Error Types
 
-In the example above, the error type is `HTTP:UNAUTHORIZED`, not simply `UNAUTHORIZED`. +
-Error types consist of both a namespace and an identifier, allowing you to distinguish the types according to their domain.
-For example, there are `HTTP:NOT_FOUND` and `FILE:NOT_FOUND` error types. +
-While connectors define their namespace, core runtime errors have the implicit `MULE` one. +
-So `MULE:EXPRESSION` and `EXPRESSION` are interpreted as one.
+Mule errors have a namespace (such as `HTTP:` and `FILE:`) and identifier
+(such as `NOT_FOUND`), and they belong to a hierarchy of error types.
 
-Another important characteristic of error types is that they may have a parent type. For example, `HTTP:UNAUTHORIZED` has `MULE:CLIENT_SECURITY` as the parent, which, in turn, has `MULE:SECURITY` as the parent. This establishes error types as specifications of more global ones: an HTTP unauthorized error is a type of client security error, which is a type of a more broad security issue.
+The naming convention enables you to
+distinguish errors according to their domain. For example, instead of
+throwing `NOT_FOUND`, the HTTP connector can throw `HTTP:NOT_FOUND`, and the
+File connector can throw `FILE:NOT_FOUND` errors.
 
-These hierarchies allow routing to be more general. For example, a handler for `MULE:SECURITY` catches HTTP unauthorized errors as well as OAuth errors. +
-Below you can see what the core runtime’s hierarchy looks like:
+Unlike connectors, Mule runtime errors have an implicit `MULE` namespace, so
+`MULE:EXPRESSION` and `EXPRESSION` are interpreted the same way.
+
+Error types can have a parent
+type. For example, the parent of `HTTP:UNAUTHORIZED` is `MULE:CLIENT_SECURITY`,
+and the parent of `MULE:CLIENT_SECURITY` is `MULE:SECURITY`. This hierarchy
+implies that each error type is a specification of more global error type.
+For example, an `HTTP:UNAUTHORIZED` error type is a type of client security
+error (`MULE:CLIENT_SECURITY`), which is a type of a more broad security issue
+(`MULE:SECURITY`).
+
+The hierarchies enable you to route errors in a general or more specific way. For
+example, an error handler for `MULE:SECURITY` catches `HTTP:UNAUTHORIZED` errors
+and OAuth errors. The follow figure shows the core runtime hierarchy:
 
 image::error-hierarchy.png[Error Hierarchy]
 
-All errors belong to one of these two main types: `ANY` or `CRITICAL`. All types under `ANY` are matched by its parent and can be handled, while error types under `CRITICAL` are so severe that cannot be handled and will only be logged.  +
+All errors belong to one of these two main types: `ANY` or `CRITICAL`. Each type
+under `ANY` is matched by its parent and can be handled, while error types under
+`CRITICAL` are so severe that cannot be handled and are only logged.
```

Comment:
Maybe give an example of a critical error?

Reply:
I'm adding that they include include `FATAL` and `OVERLOAD`. Those are described below.